src/hypothesis/_settings.py

```diff
@@ -521,18 +490,44 @@ def _validate_timeout(n):
     future_default=False,
 )
 
+
+def _validate_database(db):
+    from hypothesis.database import ExampleDatabase
+    if db is None or db is not_set or isinstance(db, ExampleDatabase):
```

Comment:
Is this sort of "database or string" logic what we were doing before? I don't think it's a good idea - if we're getting rid of the `database_file` setting and just using the `database` setting I think the database setting should always be an actual database (or `None`)

Reply:
Actually, no - we were stricter before, though (sigh) there was no explicit validation that `database` was an instance of `ExampleDatabase`.  I'll add that check, along with a very clear message for transitioning users.